src/modals/Validate.tsx

```diff
@@ -85,7 +85,7 @@ function Validate(props: ValidateProps) {
             const tempSmartWallets = [...smartWallets, smartWalletWithBalance];
             setSmartWallets(tempSmartWallets);
             localStorage.setItem(
-                state.account,
+                `${state.account}-${state.chainId}`,
```

Comment:
I think we should have function like `getAddressKey` or similar to be reused elsewhere

Reply:
updated